### PR TITLE
Recording window events

### PR DESCRIPTION
**Pull request by konfrontend**

### CODE REVIEW

**Comment by konfrontend** [comment]
[Real browser recording in Loadster Recorder](https://trello.com/c/L1p9YgV9/381-real-browser-recording-in-loadster-recorder)

File: js/windowEventRecorder.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    } catch (err) {
+        console.debug('caught error', err)
+    }
+}
+
+const recordEvent = (e) => {
+    // we explicitly catch any errors and swallow them, as none node-type events are also ingested.
+    // for these events we cannot generate selectors, which is OK
+    try {
+        const optimizedMinLength = (e.target.id) ? 2 : 10 // if the target has an id, use that instead of multiple other selectors
```
Comment (azhawkes) [inline js/windowEventRecorder.js:26]: It seemed like this wasn't working in Chrome... a couple times I got long 5-part selectors when an ID attribute was available. Can you double check that the ID is used in such cases?

File: js/windowEventRecorder.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        const selector = finder(e.target, { seedMinLength: 5, optimizedMinLength: optimizedMinLength })
+        const msg = {
+            selector: selector,
```
Comment (azhawkes) [inline js/windowEventRecorder.js:29]: A list of tag attributes might be nice here too (like `id`, `class`, `name`, etc).

**Comment by konfrontend** [comment]
@azhawkes Alright, I added a few changes, now we have more useful data. 
Thanks, this `webNavigation.onCommitted` looks promising, works great so far.

**Comment by azhawkes** [comment]
@konfrontend Cool, just ran it and I see the navigations are being detected. Can you also create pseudo-events for these navigations? Something like:

```
{ type: "loadster_browser_event", value: { action: "navigate", url: "https://loadster.app" }}
```

And then pass those, as well as the other actions (click, etc) to a handler function so they can get sent to Loadster via the `setInterval` as we are currently doing for HTTP recording events. Basically, I'm thinking the JS listening on the other end will receive both the new and old events and decide which it cares about depending on the type of recording the user triggered. Sound ok?

BTW I noticed the initial tab opening is not triggering a navigate event. Can you fake it so we record the initial navigation?

**Comment by konfrontend** [comment]
@azhawkes 
I slightly modified the sent message to send both event types together. Loadster should have backward compatibility for this(see another PR). 
Well, I think we are now ready for the UI part :)

**Comment by azhawkes** [comment]
@konfrontend Is this (still) ready to merge and publish? Any last changes before I publish the new version?

**Comment by konfrontend** [comment]
@azhawkes Yes, it is ready to merge. Nothing else comes to my mind for now. We dispatch all types of events and let vue.app to decide what to use.

**Pull request merged**